From 6615d11b0e2020aa8567eab037feb552f66428e1 Mon Sep 17 00:00:00 2001
From: Valentine Barshak <valentine.barshak@cogentembedded.com>
Date: Thu, 8 Jul 2021 01:56:09 +0300
Subject: [PATCH 2/3] arm64: dts: renesas: r8a77970: Remove redundant isp0 node

This removes isp0 node, and adds "renesas,isp-r8a77970"
to the tisp0 compatible property. Now tisp0 node should
be used instead of isp0. This prevents resource conflict
when both rcar-isp and uio_pdrv_genirq drivers are used.

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 arch/arm64/boot/dts/renesas/r8a77970.dtsi | 14 ++------------
 1 file changed, 2 insertions(+), 12 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a77970.dtsi b/arch/arm64/boot/dts/renesas/r8a77970.dtsi
index 6c7b47d..e278025 100644
--- a/arch/arm64/boot/dts/renesas/r8a77970.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a77970.dtsi
@@ -23,7 +23,6 @@
 		i2c2 = &i2c2;
 		i2c3 = &i2c3;
 		i2c4 = &i2c4;
-		isp0 = &isp0;
 		tisp0 = &tisp0;
 		cisp0 = &cisp0;
 	};
@@ -1145,18 +1144,9 @@
 			ipmmu-id = <RCAR_IPMMU_VI0 0>;
 		};
 
-		isp0: isp@fec00000 {
-			compatible = "renesas,isp-r8a77970";
-			reg = <0 0xfec00000 0 0x20000>,
-			      <0 0xfed00000 0 0x10000>;
-			interrupts = <GIC_SPI 26 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 25 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 817>;
-			power-domains = <&sysc R8A77970_PD_ALWAYS_ON>;
-		};
-
 		tisp0: tisp_00 {
-			compatible = "generic-uio";
+			compatible = "renesas,isp-r8a77970",
+				     "generic-uio";
 			reg = <0 0xfed00000 0 0x20000>;
 			interrupts = <GIC_SPI 25 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&cpg CPG_MOD 817>;
-- 
2.7.4

